assignment-2/lean.md

```diff
@@ -1,3 +1,147 @@
+waste in terms of processes and management, instead of being integrated
+within the development. In scrum, this waste does not exist because it
+is considered part of the development and it does not carry extra
+overhead due to being integrated into the development phase.
+
+In RUP, the construction phase is done by a requirements team which is
+composed by requirement engineers who are not the developers who will
+implement the software during the implementation phase. This practice is
+suboptimal because it leads to a situation in which there are
+intermediaries and artefacts between the clients and the team that will
+implement the product. In scrum, although the requirements are gathered
+by the *product owner*, the development team which will implement the
+product is involved in the creation of the backlog.
+
+On the other hand, scrum can be more wasteful than RUP. Scrum works with
+smalls teams and big projects require more than one scrum team working
+on it. This means that those teams would have each one their own *scrum
+master* who will act as facilitator and a *product owner* who will talk
+with the clients. This situation will lead to role duplicity which is
+not optimal at all. It is true that given the case, the scrum master and
+the product owner can be shared between different teams but if teams
+belong to different organisations or divisions will not happen. RUP only
+has one management team which is an improvement in terms of waste over
+scrum.
+
+Scrum also focuses on small iterations of the product: features are
+added iteration after iteration. The client then will not receive the
+final product requested during the first iterations, which is a waste
+because it is not what was requested. We understand as an *incomplete
+product*, a product which does not have all the features requested
+originally by the client. Depending on the definition of done, a
+complete product may include the testing and deployment. In addition,
+those iterations may lead to having to reimplement features due to
+changes in the architecture. This problem, which is inherited from the
+agile development philosophy, does not exist in RUP as it has only one
+implementation phase. Scrum also requires to perform testing every
+iteration for the existing codebase, which RUP does only once the whole
+project is finished as part of their quality assurance..
+
+Iterative development has the problem also that requires to deliver
+products constantly. This leads to a situation in which some operations
+need to be repeated every sprint when in RUP they only need to be
+performed once when the product is delivered. The backlog needs to be
+reviewed after every week, full testing and deployment, retrospectives…
+A constant waste that accumulates iteration after iteration of the
+partial product.
+
+Scrum might deliver a product which is not the exact one requested. The
+*product owner* discusses the details with the clients and
```

Comment:
What is the damage here? One sprint at most? Seems like a lot better than when you are not using iterations and find out after several months that you are not getting what you asked for.